helpers/kubernetes/cluster.go

```diff
@@ -260,6 +261,32 @@ func (c *Cluster) WaitForJobCompleted(namespace, jobName string, timeout time.Du
 	return wait.PollImmediate(time.Second, timeout, c.IsJobCompleted(client, jobName, namespace))
 }
 
+// IsDeploymentCompleted returns a condition function that indicates whether the given
+// Deployment is in Completed state or not.
+func (c *Cluster) IsDeploymentCompleted(deploymentName, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		deployment, err := c.Kubectl.AppsV1().Deployments(namespace).Get(context.Background(),
+			deploymentName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		for _, condition := range deployment.Status.Conditions {
+			if condition.Type == appsv1.DeploymentAvailable && condition.Status == v1.ConditionTrue {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+}
+
+func (c *Cluster) WaitForDeploymentCompleted(ui *termui.UI, namespace, deploymentName string, timeout time.Duration) error {
+	s := ui.Progressf("Starting %s in %s", deploymentName, namespace)
```

Comment:
I'm late to the party but anyway, this method could be used when updating the Deployment (thus not "Starting" but "Updating") or simply in a place where we don't want to print anything. I think the caller should decide whether to print any text or not because the caller has better context. This method shouldn't need a `termui.UI`.